pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentRestletResource.java

```diff
@@ -180,7 +185,7 @@ private Representation getAllSegmentsMetadataForTable(
   @Summary("Enable, disable or drop a segment from a table")
   @Tags({ "segment", "table" })
   @Paths({ "/tables/{tableName}/segments/{segmentName}", "/tables/{tableName}/segments/{segmentName}/" })
-  protected Representation toggleOneSegmentState(
+  public Representation toggleOneSegmentState(
```

Comment:
As discussed, please call the appropriate annotated method to handle the URIs from get(). You can then call a common method below that handles special cases.